app/models/government.rb

```diff
@@ -20,4 +22,51 @@ def self.on_date(date)
   def current?
     self == Government.current
   end
+
+  def overlaps?(other)
+    !(is_before?(other) || is_after?(other))
+  end
+
+private
+
+  def is_before?(other)
+    starts_before?(other) && ends_before?(other)
+  end
+
+  def is_after?(other)
+    starts_after?(other) && ends_after?(other)
+  end
+
+  def starts_before?(other)
+    start_date < other.start_date
+  end
+
+  def ends_before?(other)
+    (end_date || distant_future) < other.start_date
+  end
+
+  def starts_after?(other)
+    start_date > (other.end_date || distant_future)
+  end
+
+  def ends_after?(other)
+    (end_date || distant_future) > (other.end_date || distant_future)
+  end
+
+  def not_overlapping?
+    self.class.all.each do |existing_government|
+      next if self == existing_government
+
+      if self.overlaps?(existing_government)
+        errors.add(:base, "overlaps #{existing_government.name}:
+          This government: #{self.start_date} -> #{self.end_date || distant_future},
+          existing government: #{existing_government.start_date} -> #{existing_government.end_date || distant_future}
```

Comment:
I feel like "existing government" is a loaded term here - it could be read as the current government, and the one that this government is overlapping with isn't necessarily the current one. How about "other government" instead?